core/Pos/Core/Genesis/Types.hs

```diff
@@ -57,20 +48,44 @@ getTotalStake ExponentialStakes = mkCoin . sum $
     let g 0 = []
         g n = n : g (n `div` 2)
     in g 5000
-getTotalStake (ExplicitStakes balances) = unsafeIntegerToCoin $
-    sumCoins $ fst <$> balances
-getTotalStake (CombinedStakes st1 st2) =
-    getTotalStake st1 `unsafeAddCoin` getTotalStake st2
+getTotalStake (CustomStakes balances) =
+    unsafeIntegerToCoin $ sumCoins balances
 
--- | Hardcoded genesis data
-data GenesisCoreData = GenesisCoreData
-    { gcdAddresses         :: [Address]
-    , gcdDistribution      :: StakeDistribution
-    , gcdBootstrapBalances :: !(HashMap StakeholderId Coin)
+-- | List of distributions accompained by related addresses set (what
+-- to distribute and how).
+type AddrDistribution = [(HashSet Address, StakeDistribution)]
```

Comment:
Yes, it's a bug probably. I will replace this datatype with a list and ensure it doesn't have duplicates in `mkGenesisCoreData`.